packages/@lwc/engine-core/src/framework/api.ts

```diff
@@ -49,6 +51,18 @@ function addVNodeToChildLWC(vnode: VCustomElement) {
     ArrayPush.call(getVMBeingRendered()!.velements, vnode);
 }
 
+// [st]atic node
+function st(elmProto: Element, key: Key): VStatic {
```

Comment:
why not reusing VElementData here? and take the key out of the data? My hunch is that eventually we might be able to do some crazy optimizations where the data payload can provide some stuff for the subtree, so it makes sense to keep it around.

Reply:
atm is not needed and this is internals that we could modify to add the data bag once we have an optimization in mind. passing the key directly will reduce size (a small amount), and perf (destructing key from the databag). however, I can generate the databag that's preferred.